Configuration/Skimming/interface/DisappearingMuonsSkimming.h

```diff
@@ -47,28 +49,28 @@ class DisappearingMuonsSkimming : public edm::one::EDFilter<> {
   static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
 
 private:
-  void beginJob() override;
   bool filter(edm::Event&, const edm::EventSetup&) override;
-  void endJob() override;
+
   bool passTriggers(const edm::Event& iEvent,
-                    edm::EDGetToken m_trigResultsToken,
+                    const edm::TriggerResults& results,
                     std::vector<std::string> m_muonPathsToPass);
-  double getTrackIsolation(const edm::Event&,
-                           edm::Handle<reco::VertexCollection> vtxHandle,
-                           std::vector<reco::Track>::const_iterator& iTrack);
+
+  bool findTrackInVertices(const reco::TrackRef& tkToMatch,
+                           const reco::VertexCollection& vertices,
+                           unsigned int& vtxIndex,
+                           unsigned int& trackIndex);
+
+  double getTrackIsolation(const reco::TrackRef& tkToMatch, const reco::VertexCollection& vertices);
   double getECALIsolation(const edm::Event&, const edm::EventSetup&, const reco::TransientTrack track);
```

Comment:
```suggestion
  double getECALIsolation(const edm::Event&, const edm::EventSetup&, const reco::TransientTrack& track);
```
Can we pass `track` by reference?

Reply:
yes done in https://github.com/cms-sw/cmssw/pull/41225/commits/7a3a15df8d7e392f2e1394ca2120e5c5f2e6512c